spec/requests/api/v1/countries/create_spec.rb

```diff
@@ -0,0 +1,47 @@
+describe 'POST /api/v1/countries', type: :request do
+  subject { post api_v1_countries_path, headers: headers, params: params, as: :json }
+
+  let(:headers) { auth_headers }
+  let(:user) { create(:user) }
+  let(:country_name) { Faker::Address.country }
+  let(:params) do
+    {
+      country: {
+        name: country_name
+      }
+    }
+  end
+
+  before { subject }
```

Comment:
I don't like of creating `before` blocks to call `subject`. So I'd say:
- If you need the subject to run before an expectation but there is another `it` inside the same context that does not need that, call `subject` explicitly in the `it` block
```ruby
it 'tests something' do
  subject
  # my_expectation_here
end
```
- if you have a context where you need to call `subject` for **every** `it` block, you can define the `subject` with a bang (`subject!`)
```ruby
context 'where all my tests run after calling subject' do
  subject! { anything }

  it 'tests side effects' do
    # expects here
  end
end
```

Reply:
Updated